### PR TITLE
add charliecloud 0.16

### PR DESCRIPTION
**Pull request by j-ogas**

### CODE REVIEW

**Review by alalazo — APPROVED** [review]
@reidpr I'll merge this if you agree being a maintainer of Spack's `charliecloud` package. This just means that if there are changes to this recipe in the future we'll ping you for reviews.

**Review by adamjstewart — APPROVED** [review]

**Comment by reidpr** [comment]
Yes, I agree. Thank you.

**Pull request merged**
